bindings/python/src/layers.rs

```diff
@@ -92,14 +71,18 @@ impl RetryLayer {
         if let Some(min_delay) = min_delay {
             retry = retry.with_min_delay(Duration::from_micros((min_delay * 1000000.0) as u64));
         }
-        Ok(Self(retry))
+
+        let retry_layer = Self(retry);
+        let class = PyClassInitializer::from(Layer(Box::new(retry_layer.clone())))
+            .add_subclass(retry_layer);
+
+        Ok(class)
     }
 }
 
 pub fn create_submodule(py: Python) -> PyResult<&PyModule> {
     let submod = PyModule::new(py, "layers")?;
-    submod.add_class::<ConcurrentLimitLayer>()?;
-    submod.add_class::<ImmutableIndexLayer>()?;
+    submod.add_class::<Layer>()?;
```

Comment:
I wonder that happens when user subclass `Layer` in Python, maybe we shouldn't export `Layer` type to Python just yet?

Reply:
> I wonder that happens when user subclass `Layer` in Python

Yes, we should forbid users to subclass `Layer`.

> maybe we shouldn't export `Layer` type to Python just yet?

I made it public since it's part of our API:

```python
class AsyncOperator:
    def __init__(self, scheme: str, **kwargs): ...
    def layer(self, layer: Layer): ...
```

Can we expose this API but not expose `Layer` class?

Reply:
> Yes, we should forbid users to subclass `Layer`.

In fact, I'm not sure about this. Can users from python can implement this trait?

Reply:
> In fact, I'm not sure about this. Can users from python implement this trait?

I think they can't because `Layer` lacks `#[new]` so it can't be constructed from Python, so probably fine to leave it as-is.

Reply:
Got it, thanks for the review!